### PR TITLE
Add support for creating Freezer Manager freezer hierarchies via StorageController APIs

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
In the related PR for this epic we added a new StorageController with actions for create/update/delete of freezer hierarchy items (i.e. freezer physical locations, freezers, locations within the freezer, storage unit types, and terminal storage locations). This PR adds a new namespace (storage) to the Java API with commands for calling those StorageController actions.

#### Related Pull Requests
* https://github.com/LabKey/inventory/pull/542
* https://github.com/LabKey/inventory/pull/545
* https://github.com/LabKey/sampleManagement/pull/1221

#### Changes
* storage namespace with CreateCommand, UpdateCommand, DeleteCommand
* JavaDocs and examples

### CODE REVIEW

File: labkey-client-api/src/org/labkey/remoteapi/storage/CreateCommand.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * Create a new LabKey Freezer Manager storage item which can then be combined to create a freezer hierarchy.
+ * Freezer hierarchies consist of a top level freezer which can have any combination of child non-terminal storage
+ * locations (i.e. those that do not directly contain samples but can contain other units) and terminal storage
+ * locations (i.e. units in the freezer which directly contain samples and cannot contain other units).
+ * See the <a href="https://www.labkey.org/SampleManagerHelp/wiki-page.view?name=createFreezer">LabKey Documentation</a>
+ * for further details.
+ * </p><p>
+ * A freezer may also have a parent hierarchy which defines the physical location of the freezer.
+ * See the <a href="https://www.labkey.org/SampleManagerHelp/wiki-page.view?name=freezerLocation">LabKey Documentation</a>
+ * for further details.
```
Comment (labkey-susanh) [inline labkey-client-api/src/org/labkey/remoteapi/storage/CreateCommand.java]:
```suggestion
 * Create a new LabKey Freezer Manager storage item that can be used in the creation of a freezer hierarchy.
 * Freezer hierarchies consist of a top level Freezer, which can have any combination of child non-terminal storage
 * locations (i.e. those that do not directly contain samples but can contain other units) and terminal storage
 * locations (i.e. units in the freezer that directly contain samples and cannot contain other units).
 * See the <a href="https://www.labkey.org/SampleManagerHelp/wiki-page.view?name=createFreezer">LabKey Documentation</a>
 * for further details.
 * </p><p>
 * A freezer may also have a parent hierarchy, which defines the physical location of the freezer.
 * See the <a href="https://www.labkey.org/SampleManagerHelp/wiki-page.view?name=freezerLocation">LabKey Documentation</a>
 * for further details.
```

File: labkey-client-api/src/org/labkey/remoteapi/storage/DeleteCommand.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
+package org.labkey.remoteapi.storage;
+
+/**
+ * <p>
+ * Delete an existing LabKey Freezer Manager storage item. Note that deletion of freezers or locations within the
+ * freezer hierarchy will cascade the delete down the hierarchy to remove child locations and terminal storage locations.
```
Comment (labkey-susanh) [inline labkey-client-api/src/org/labkey/remoteapi/storage/DeleteCommand.java:5]: Do we want to make any note here about the behavior when samples are in storage?

File: labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
+package org.labkey.remoteapi.storage;
+
+import org.json.simple.JSONObject;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class StorageRow
```
Comment (labkey-susanh) [inline labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java:12]: Javadoc for the class?

File: labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+{
+    private String _type;
```
Comment (labkey-susanh) [inline labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java]: Is it worth making an enum for this so users don't need to rely on the docs to find out valid values?

**Review by labkey-susanh — APPROVED** [review]

File: labkey-client-api/src/org/labkey/remoteapi/storage/DeleteCommand.java (hunk 2 above)
Comment (cnathe) [reply labkey-client-api/src/org/labkey/remoteapi/storage/DeleteCommand.java:5]: done

File: labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java (hunk 3 above)
Comment (cnathe) [reply labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java:12]: done

File: labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java (hunk 4 above)
Comment (cnathe) [reply labkey-client-api/src/org/labkey/remoteapi/storage/StorageRow.java]: done

**Pull request merged**